service/controller/resource/monitoring/verticalpodautoscaler/resource.go

```diff
@@ -140,12 +143,19 @@ func (r *Resource) getObject(ctx context.Context, v interface{}) (*vpa_types.Ver
 	return vpa, nil
 }
 
-func (r *Resource) getMaxCPU(ctx context.Context) (*resource.Quantity, error) {
-	nodes, err := r.k8sClient.K8sClient().CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+func (r *Resource) listWorkerNodes(ctx context.Context) (*v1.NodeList, error) {
```

Comment:
I would have moved this to a shared place because I see this being reusable but I'm fine for now